sockeye/utils.py

```diff
@@ -264,6 +264,21 @@ def average_tensors(tensors: List[pt.Tensor]) -> pt.Tensor:
     return sum(tensors) / len(tensors)  # type: ignore
 
 
+def one_hot_encoding_from_prefix(prefix: pt.Tensor, vocab_size: int) -> pt.Tensor:
+    """
+    Create an one hot encoding from output vocab size of prefix
+
+    :param prefix: Target prefix ids. Shape (batch size, 1).
+    :param vocab_size: vocabulary size
+    :return prefix_in_one_hot (batch size, 1, output_vocab_size)
+
+    """
+    prefix_in_one_hot = pt.nn.functional.one_hot(prefix.to(pt.int64), num_classes=vocab_size)
```

Comment:
This is interesting. I am not that sure (not clear in the document from pytorch)... To be safe let us also assign the device of prefix_in_one_hot to prefix.device as well to avoid any potential error.